File: php/leetcode/arrays.php
<?php
declare(strict_types=1);
// TOPIC: Array LeetCode Problems | php arrays.php

// TODO 1: Two Sum — LC #1
//   function twoSum(array $nums, int $target): array
//   // O(n) hash map: $complement = $target - $num; return [$map[$complement], $i] if found

// TODO 2: Best Time to Buy and Sell Stock — LC #121
//   function maxProfit(array $prices): int
//   // Track min price; $maxProfit = max($maxProfit, $price - $minPrice)

// TODO 3: Container With Most Water — LC #11
//   function maxArea(array $height): int
//   // Two pointers; advance the shorter side

// TODO 4: Product of Array Except Self — LC #238
//   function productExceptSelf(array $nums): array
//   // Prefix + suffix product arrays; no division

// TODO 5: Maximum Subarray (Kadane's) — LC #53
//   function maxSubArray(array $nums): int
//   // $current = max($num, $current + $num); track max

// TODO 6: Merge Intervals — LC #56
//   function merge(array $intervals): array
//   // usort by start; merge if $intervals[$i][0] <= $prev[1]

// TODO 7: Find Minimum in Rotated Sorted Array — LC #153
//   function findMin(array $nums): int
//   // Binary search: if $nums[$mid] > $nums[$right], min is in right half

File: ruby/leetcode/arrays.rb
# frozen_string_literal: true
# TOPIC: Array LeetCode Problems | ruby arrays.rb
# Docs: https://leetcode.com/tag/array/

# TODO 1: Two Sum — LC #1
#   def two_sum(nums, target)
#     # O(n) hash map: complement = target - num; return [map[complement], i] if found
#   end

# TODO 2: Best Time to Buy and Sell Stock — LC #121
#   def max_profit(prices)
#     # Track min_price seen so far; max_profit = max(profit, price - min_price)
#   end

# TODO 3: Container With Most Water — LC #11
#   def max_area(height)
#     # Two pointers: left=0, right=last; move the shorter side inward
#   end

# TODO 4: Product of Array Except Self — LC #238
#   def product_except_self(nums)
#     # Pass 1: prefix products left→right; Pass 2: suffix products right→left; no division
#   end

# TODO 5: Maximum Subarray (Kadane's) — LC #53
#   def max_sub_array(nums)
#     # current_sum = max(num, current_sum + num); track max
#   end

# TODO 6: Rotate Array — LC #189
#   def rotate(nums, k)
#     # Reverse whole array, reverse first k, reverse remaining (in-place)
#   end

# TODO 7: Merge Intervals — LC #56
#   def merge(intervals)
#     # Sort by start; iterate and merge overlapping [a,b] + [c,d] where c <= b → [a, max(b,d)]
#   end

# TODO 8: Find Minimum in Rotated Sorted Array — LC #153
#   def find_min(nums)
#     # Binary search: if nums[mid] > nums[right], min is in right half
#   end
